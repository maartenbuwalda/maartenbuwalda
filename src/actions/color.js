export const SET_DOMINANT_COLOR = 'SET_DOMINANT_COLOR'
export const GET_DOMINANT_COLOR = 'GET_DOMINANT_COLOR'

// export const setDominantColor = palette => {
//   return {
//     type: SET_DOMINANT_COLOR,
//     payload: {
//       colorRegular: palette.color,
//       colorLight: palette.colorLight,
//       colorDark: palette.colorDark,
//     }
//   }
// }

export const getDominantColor = image => {
  return {
    type: GET_DOMINANT_COLOR,
    payload: {
      image
    }
  }
}
