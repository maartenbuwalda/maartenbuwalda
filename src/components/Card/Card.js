import React, { Component } from 'react'
import styles from './Card.sass'

class Card extends Component {
  render() {
    console.log(this.props)
    return (
      <div style={{backgroundColor: "rgb(" + this.props.color.toString() + ")"}} className={styles["card"]}>
        <div 
          className={styles["card-image"]}
          style={{backgroundImage: 'url(' + this.props.content.image + ')'}}/>
      	<h2>{this.props.content.title}</h2>
      	<p>{this.props.content.body}</p>
      </div>
    );
  }
}

export default Card
