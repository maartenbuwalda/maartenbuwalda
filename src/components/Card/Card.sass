.card
  background-color: white
  // padding: 1rem
  // max-width: 20rem
  min-height: 15rem
  
  width: 100%
  max-width: 20rem
  margin: 1rem
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);

  figure
    margin: 0
    width: 100%
    height: 15rem
    overflow: hidden

    img
      display: block
      max-width: 100%

  h2
    margin: 1rem
    transition: all 0.5s cubic-bezier(.25,.8,.25,1)

  p
    margin: 1rem