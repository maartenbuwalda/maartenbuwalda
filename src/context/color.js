import React from 'react'

export const colors = {
  hex: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  },
  rgb: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  }
}

export const ColorContext = React.createContext({
  theme: colors,
  updateTheme: () => {},
})
