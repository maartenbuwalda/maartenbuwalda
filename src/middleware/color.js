import {
  GET_DOMINANT_COLOR,
  SET_DOMINANT_COLOR,
} from '../actions/color'

import Vibrant from 'node-vibrant'

const color = store => next => action => {
  next(action)
  const dispatch = store.dispatch

  if (action.type === GET_DOMINANT_COLOR) {
    const img = new Vibrant(action.payload.image)
    img.getPalette((err, palette) => {
      if (!err) {
        dispatch({
          type: SET_DOMINANT_COLOR,
          payload: {
            hex: {
              colorMutedDark: palette.DarkMuted && palette.DarkMuted.getHex(),
              colorMutedLight: palette.LightMuted && palette.LightMuted.getHex(),
              colorMutedRegular: palette.Muted && palette.Muted.getHex(),
              colorDark: palette.DarkVibrant && palette.DarkVibrant.getHex(),
              colorLight: palette.LightVibrant && palette.LightVibrant.getHex(),
              colorRegular: palette.Vibrant && palette.Vibrant.getHex(),
            },
            rgb: {
              colorMutedDark: palette.DarkMuted && palette.DarkMuted.getRgb().toString(),
              colorMutedLight: palette.LightMuted && palette.LightMuted.getRgb().toString(),
              colorMutedRegular: palette.Muted && palette.Muted.getRgb().toString(),
              colorDark: palette.DarkVibrant && palette.DarkVibrant.getRgb().toString(),
              colorLight: palette.LightVibrant && palette.LightVibrant.getRgb().toString(),
              colorRegular: palette.Vibrant && palette.Vibrant.getRgb().toString(),
            }
          }
        })
      }
    })
  }
}

export default color
