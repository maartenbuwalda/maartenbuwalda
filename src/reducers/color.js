import {
  SET_DOMINANT_COLOR,
  GET_DOMINANT_COLOR,
} from '../actions/color'

const defaultState = {
  isFetching: false,
  hex: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  },
  rgb: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  }
}

function colorReducer (state = defaultState, action) {
  switch (action.type) {
    case GET_DOMINANT_COLOR:
      return {
        ...state,
        isFetching: true,
      }
    case SET_DOMINANT_COLOR:
      return {
        ...state,
        isFetching: false,
        hex: action.payload.hex,
        rgb: action.payload.rgb,
      }
    default:
      return state
  }
}

export { colorReducer }
