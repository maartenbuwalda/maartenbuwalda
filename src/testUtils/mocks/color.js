export const color = {
  hex: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  },
  rgb: {
    colorMutedRegular: '',
    colorMutedLight: '',
    colorMutedDark: '',
    colorRegular: '',
    colorLight: '',
    colorDark: '',
  }
}
