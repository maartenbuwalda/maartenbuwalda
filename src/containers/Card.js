import Card from '../components/Card'
import { connect } from 'react-redux'

const mapStateToProps = state => {
  return {
    color: state.color,
  }
}

export default connect(mapStateToProps)(Card)
