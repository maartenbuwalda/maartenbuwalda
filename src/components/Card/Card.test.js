import React from 'react'
import { mount } from 'enzyme'
import Card from './index'
import { componentRenders } from '../../testUtils/rendering'
import { color } from '../../testUtils/mocks/color'

describe('Card component', () => {
  const component = <Card />
  componentRenders(component)

  describe('when rendering with props', () => {
    const component = <Card color={color} position="test" />
    const mounted = mount(component)
    it('renders correctly', () => {
      expect(mounted)
    })
  })

  describe('when rendering with a transparent background', () => {
    const component = <Card transparent color={color} position="test" />
    const mounted = mount(component)
    it('renders correctly', () => {
      expect(mounted)
    })
  })
})
