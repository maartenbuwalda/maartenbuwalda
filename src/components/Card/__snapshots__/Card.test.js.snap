// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<styled.article>
  <styled.main />
</styled.article>
`;
